src/modules/textFacets/__tests__/textFacets.test.js

```diff
@@ -0,0 +1,152 @@
+import renderer from 'react-test-renderer';
+import { render, waitFor, fireEvent } from '@testing-library/react';
+import React from 'react';
+import Products from '../../products/index';
+import TextFacets from '../index';
+import UnbxdSearchWrapper from '../../../UnbxdSearchWrapper';
+import SearchBox from '../../searchBox';
+import { searchResponse } from './mocks/searchMock';
+
+const FacetItemComponent = ({ itemData, onClick }) => {
+    const { name, count, isSelected } = itemData;
+    const handleClick = () => {
+        onClick(itemData);
+    };
+
+    return (
+        <div
+            className={`UNX-facet__item ${isSelected ? '-selected' : ''}`}
+            onClick={handleClick}
+        >
+            <div className="-checkbox" />
+            <div className="-label">{name}</div>
+            <div className="-count">({count})</div>
+        </div>
+    );
+};
+
+// establish API mocking before all tests
+beforeAll(() => {
+    window.fetch = jest.fn(() => {
+        return Promise.resolve({
+            json: () => Promise.resolve(searchResponse),
+        });
+    });
+})
+
+const attributesMap = {
+    productName: 'title',
+    uniqueId: 'uniqueId',
+    imageUrl: 'imageUrl',
+    price: 'min_cheapest_default_price',
+    productUrl: 'productUrl'
+};
+
+test('Match Snapshot for text facets', () => {
+    const tree = renderer
+        .create(
+            <UnbxdSearchWrapper
+                siteKey="wildearthclone-neto-com-au808941566310465"
+                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
+            >
+                <Products attributesMap={attributesMap} />
```

Comment:
Please use textfacet in the snapshot or else it's just a duplicate of product snapshot testing. Also, look why a snapshot is not created completely with products. Try to create a snapshot once the promise is resolved.

Reply:
working on it

Reply:
done